src/HealthGPS/analysis_module.cpp

```diff
@@ -304,6 +305,47 @@ DALYsIndicator AnalysisModule::calculate_dalys(Population &population, unsigned
                           .disability_adjusted_life_years = yll + yld};
 }
 
+void AnalysisModule::calculate_population_statistics(RuntimeContext &context) const {
+    auto num_factors_to_calculate =
+        context.mapping().entries().size() - factors_to_calculate_.size();
+
+    for (const auto &entity : context.population()) {
+        // Get factors_to_calculate_ values for this entity
+        std::vector<double> entity_factors;
+        for (const auto &factor : factors_to_calculate_) {
+            entity_factors.push_back(entity.get_risk_factor_value(factor));
+        }
+        // Get the bin index for each factor
+        std::vector<int> bin_indices;
+        for (size_t i = 0; i < factors_to_calculate_.size(); i++) {
+            auto factor_value = entity_factors[i];
+            auto bin_index =
+                static_cast<int>((factor_value - factor_bin_widths_[i]) / factor_bin_widths_[i]);
```

Comment:
I'm sure I'm missing something here, but why are you subtracting `factor_bin_widths_[i]`? Shouldn't we be subtracting the minimum possible factor value? Or do they all start at 0?

Reply:
I'm a bit puzzled by this also. Isn't `factor_bin_widths` a real number `r >= 1.0`? Surely `factor_value` would also need normalising before use?

Reply:
Eventually, given the fixed window of bins, we will need to make sure all our risk factors have config supplied boundaries, which are always respected. Then we can just read `min_value` and `max_value` directly from that and always be safe. But another time...

Reply:
Good point and well spotted. I have no idea why I had it like originally. Likely because at some point I changed the logic and missed it.